### PR TITLE
Adapter Failed to Load #958

### PR DESCRIPTION
**Pull request by DineshChirnanchu**
No errors reported for adapters that are not loaded because of 'copied from internet' bit on adapter binary

#958

### CODE REVIEW

**Comment by DineshChirnanchu** [comment]
@mayankbansal018  @singhsarab 

Pushed fixes for failing tests, kindly verify and  share your feedback

File: src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/TestRequestHandler.cs [hunk 1]
```diff
@@ -240,10 +240,11 @@ public void OnMessageReceived(object sender, MessageReceivedEventArgs messageRec
                     {
                         EqtTrace.Info("Discovery Session Initialize.");
                         this.testHostManagerFactoryReady.Wait();
+                        var discoveryEventsHandler = new TestDiscoveryEventHandler(this);
                         var pathToAdditionalExtensions = this.dataSerializer.DeserializePayload<IEnumerable<string>>(message);
                         jobQueue.QueueJob(
                                 () =>
-                                testHostManagerFactory.GetDiscoveryManager().Initialize(pathToAdditionalExtensions), 0);
+                                testHostManagerFactory.GetDiscoveryManager().Initialize(pathToAdditionalExtensions, discoveryEventsHandler), 0);
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/TestRequestHandler.cs]: >discoveryEventsHandler [](start = 116, length = 22)

should be testExtensionInitializeEventsHandler

**Comment by DineshChirnanchu** [comment]
@singhsarab , @mayankbansal018  

Overriding the  TestMessageLevel From **Error** to **Warning** in ExecutionManager.cs and DiscoveryManager.cs files to get through the build checks.
Added resource string for error message 
And the changes above are working well in local environment

File: src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/TestRequestHandler.cs (hunk 1 above)
Comment (DineshChirnanchu) [reply src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/TestRequestHandler.cs]: I will go ahead with this once we finalize the permanent fix for it

**Comment by DineshChirnanchu** [comment]
Pushed changes  @singhsarab  @mayankbansal018 
-Added new tests
-Updated Error message

File: src/Microsoft.TestPlatform.Common/Constants.cs [hunk 2]
```diff
@@ -1,8 +1,11 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+
+
 namespace Microsoft.VisualStudio.TestPlatform.Common
 {
+    using System.Collections.Generic;
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.Common/Constants.cs]: this seems unnecessary.

File: src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscovererEnumerator.cs [hunk 3]
```diff
@@ -316,12 +316,12 @@ private static void LogWarningOnNoTestsDiscovered(IEnumerable<string> sources, s
             }
             else
             {
+                //Should be logged only when tests found - if adapter failed to load then there would be no tests to process
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscovererEnumerator.cs]: revert this

File: src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs [hunk 4]
```diff
@@ -28,16 +30,40 @@ namespace Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Execution
...
         /// <summary>
         /// Initializes a new instance of the <see cref="ExecutionManager"/> class.
         /// </summary>
         public ExecutionManager(IRequestData requestData) : this(TestPlatformEventSource.Instance, requestData)
         {
+            this.sessionMessageLogger = TestSessionMessageLogger.Instance;
+            this.sessionMessageLogger.TestRunMessage += this.TestSessionMessageHandler;
+        }
+
+        private void TestSessionMessageHandler(object sender, TestRunMessageEventArgs e)
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs]: move private to bottom

File: src/Microsoft.TestPlatform.Common/Constants.cs (hunk 2 above)
Comment (DineshChirnanchu) [reply src/Microsoft.TestPlatform.Common/Constants.cs]: has been removed

File: src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscovererEnumerator.cs (hunk 3 above)
Comment (DineshChirnanchu) [reply src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscovererEnumerator.cs]: reverted

File: src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs (hunk 4 above)
Comment (DineshChirnanchu) [reply src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs]: moved under private methods region

File: src/Microsoft.TestPlatform.Common/Constants.cs [hunk 5]
```diff
@@ -1,6 +1,8 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+
+
 namespace Microsoft.VisualStudio.TestPlatform.Common
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.Common/Constants.cs]: why is this file changed

File: src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscoveryManager.cs [hunk 6]
```diff
@@ -247,6 +249,11 @@ private void TestSessionMessageHandler(object sender, TestRunMessageEventArgs e)
 
             if (this.testDiscoveryEventsHandler != null)
             {
+                if (e.Level == TestMessageLevel.Error)
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscoveryManager.cs]: >Error [](start = 48, length = 5)

Do not downgrade error to warnings, this handler is also invoked when Adapters(Extensions) send something back as error, we cannot reduce their messages as warning.

File: src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs [hunk 7]
```diff
@@ -88,6 +93,8 @@ public void Initialize(IEnumerable<string> pathToAdditionalExtensions)
             ITestCaseEventsHandler testCaseEventsHandler,
             ITestRunEventsHandler runEventsHandler)
         {
+            //unsubscrive session logger
+            this.sessionMessageLogger.TestRunMessage -= this.TestSessionMessageHandler;
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs]: this should be done in Initialize itself

File: src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs [hunk 8]
```diff
@@ -215,14 +223,37 @@ private void InitializeDataCollectors(string runSettings, ITestEventsPublisher t
...
             {
                 var inProcDataCollectionExtensionManager = new InProcDataCollectionExtensionManager(runSettings, testEventsPublisher, defaultCodeBase, TestPluginCache.Instance);
             }
         }
 
+        private void TestSessionMessageHandler(object sender, TestRunMessageEventArgs e)
+        {
+            if (this.testMessageEventsHandler != null)
+            {
+                if (e.Level == TestMessageLevel.Error)
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs]: same as discovery handler

File: test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Discovery/DiscoveryManagerTests.cs [hunk 9]
```diff
@@ -221,6 +226,25 @@ public void DiscoverTestsShouldCollectMetrics()
...
+        public void DiscoveryInitializeShouldVerifyWarningMessageIfAdapterFailedToLoad()
+        {
+            var assemblyLocation = typeof(DiscoveryManagerTests).GetTypeInfo().Assembly.Location;
+            var mockLogger = new Mock<ITestDiscoveryEventsHandler2>();
+            TestPluginCacheTests.SetupMockExtensions(
+                new string[] { assemblyLocation },
+                () => { });
+
+            //Act
+            this.discoveryManager.Initialize(new List<string> { assemblyLocation }, mockLogger.Object);
```
Comment (mayankbansal018) [inline test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Discovery/DiscoveryManagerTests.cs]: >assemblyLocation [](start = 64, length = 16)

make the message as appropriate

File: src/Microsoft.TestPlatform.Common/Constants.cs (hunk 5 above)
Comment (DineshChirnanchu) [reply src/Microsoft.TestPlatform.Common/Constants.cs]: reverted

File: src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscoveryManager.cs (hunk 6 above)
Comment (DineshChirnanchu) [reply src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscoveryManager.cs]: Removed Level downgrade lines, and now it follows whatever the Level it gets.

File: src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs (hunk 8 above)
Comment (DineshChirnanchu) [reply src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs]: Removed Level downgrade lines, and now it follows whatever the Level it gets.

File: src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs (hunk 7 above)
Comment (DineshChirnanchu) [reply src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs]: Moved in to the Initialize()

File: test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Discovery/DiscoveryManagerTests.cs (hunk 9 above)
Comment (DineshChirnanchu) [reply test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Discovery/DiscoveryManagerTests.cs]: Removed unnecessary test and updated message.

**Comment by DineshChirnanchu** [comment]
@mayankbansal018 , suggested changes pushed, kindly have a look

**Review by mayankbansal018 — APPROVED** [review]
:shipit:

**Comment by DineshChirnanchu** [comment]
@mayankbansal018  

Reverted unnecessary changes.
Verified 'No tests available' message behavior in old one and mine seems it is same, so issue with that.
Left the ExecutionManager unit test commented for future reference.
